### PR TITLE
Fix token creation without comment or expiration

### LINKED ISSUE #494: [ISSUE] Error: requirement failed: Invalid timeout: 0 on token creation
### Terraform Version
Terraform 0.14.6
Databricks provider 0.3.0

### Affected Resource(s)
- databricks_token

### Terraform Configuration Files
```hcl
resource "databricks_token" "token" {}
```

### Debug Output
Cannot be disclosed publicly, mail me at _sam.debruyn at lynxcare.eu_

### Expected Behavior
Token created like it did in previous versions

### Actual Behavior
```
databricks_token.token: Creating...

Error: requirement failed: Invalid timeout: 0
Please consult API docs at https://docs.databricks.com/dev-tools/api/latest/token.html#create for details.
```

### PR DESCRIPTION
**Pull request by sdebruyn**
fixes #494 

- allow tokens without comment or expiration
- add acceptance test for tokens without comment & expiration

### CODE REVIEW

File: identity/resource_token.go [hunk 1]
```diff
@@ -50,10 +50,16 @@ type TokensAPI struct {
...
 func (a TokensAPI) Create(tokenLifetime time.Duration, comment string) (r TokenResponse, err error) {
-	err = a.client.Post(a.context, "/token/create", TokenRequest{
-		LifetimeSeconds: int32(tokenLifetime.Seconds()),
-		Comment:         comment,
-	}, &r)
+	request := TokenRequest{}
+	seconds := int32(tokenLifetime.Seconds())
+	if seconds > 0 {
+		request.LifetimeSeconds = seconds
+	}
```
Comment (nfx) [inline identity/resource_token.go:57]: can you introduce a constant in the top of the file, `DefaultTokenLifetimeSeconds`, and make it to 3600 and reference it here?...

File: identity/resource_token.go [hunk 2]
```diff
@@ -98,13 +104,11 @@ func ResourceToken() *schema.Resource {
 			Type:     schema.TypeInt,
 			Optional: true,
 			ForceNew: true,
-			Default:  0,
 		},
 		"comment": {
 			Type:     schema.TypeString,
 			Optional: true,
 			ForceNew: true,
-			Default:  "",
```
Comment (nfx) [inline identity/resource_token.go:107]: Maybe just make it "Terraform" by default and have no if/else in the code?...

**Review by nfx — CHANGES_REQUESTED** [review]
small comments regarding default values.

File: identity/resource_token.go (hunk 1 above)
Comment (sdebruyn) [reply identity/resource_token.go:57]: I prefer not to.  How would you be able to create tokens without an expiration then? Seconds == 0 should mean that the token doesn't expire?

Comment (nfx) [reply identity/resource_token.go:57]: I think it will fail, no?

Comment (sdebruyn) [reply identity/resource_token.go:57]: No, I posted the API docs in #494, you should be able to create tokens without comments and without expiration

Comment (nfx) [reply identity/resource_token.go:57]: @sdebruyn Oh, okay. then i'll ask for one last thing - can you add `TestResourceTokenCreate_NoExpiration`, that would verify it on the resource level. I just suspect we'll have the same issue as with single node clusters. When i'll see that one new unit test (you've added integration test) with creating token with no arguments - i'll merge :)

File: identity/resource_token.go (hunk 2 above)
Comment (sdebruyn) [reply identity/resource_token.go:107]: As discussed above, empty comments should be allowed

**Review by nfx — APPROVED** [review]

**Comment by nfx** [comment]
@sdebruyn can you rebase it on master? or merge in the master? then good to go :)

**Comment by sdebruyn** [comment]
@nfx done :)

**Pull request merged**